chronicles/textlineserializer.nim

```diff
@@ -17,22 +26,47 @@ serializationFormat TextLineLog,
                     mimeType = "text/plain"
 
 #
-# Class creation
+# color support functions
 #
-proc init*(T: type TextLineWriter, stream: OutputStream): T =
-  result.stream = stream
+proc fgColor(writer: TextLineWriter, color: ForegroundColor, brightness: bool) =
+  when writer.colorScheme == AnsiColors:
+    writer.stream.writeText ansiForegroundColorCode(color, brightness)
+  when writer.colorScheme == NativeColors:
+    writer.stream.setForegroundColor(color, brightness)
+
+proc resetColors(writer: TextLineWriter) =
+  when writer.colorScheme == AnsiColors:
+    writer.stream.writeText ansiResetCode
+  when writer.colorScheme == NativeColors:
+    writer.stream.resetAttributes()
+
+proc applyStyle(writer: TextLineWriter, style: Style) =
+  when writer.colorScheme == AnsiColors:
+    writer.stream.writeText ansiStyleCode(style)
+  when writer.colorScheme == NativeColors:
+    writer.stream.setStyle({style})
 
 #
-# Field Handling
+# Class startup
 #
+proc init*(w: var TextLineWriter, stream: OutputStream) =
+  w.stream = stream
 
+#
+# Field Handling
+#
 proc writeFieldName*(w: var TextLineWriter, name: string) =
   w.stream.writeText ' '
+  let (color, bright) = levelToStyle(w.currentLevel)
+  w.fgColor(color, bright)
   w.stream.writeText name
+  w.resetColors()
   w.stream.writeText "="
 
 proc writeValue*(w: var TextLineWriter, value: auto) =
-  w.stream.writeText value
+  w.fgColor(propColor, true)
+  w.stream.writeTextQuoted(value, optional=true)
```

Comment:
Only strings should have quotes though.

Admittedly, the format is not very well specified, but we can use `{}` as delimiters for objects, quotes for strings, `[]` for lists and arrays and normal literals for all integers and floats.

Reply:
well, the text quoting is/used to be well specified: quote-if-needed basically - ie avoid quotes ("noise") but retain parse:ability - ie optimize for humans but give machines a chance

Reply:
@arnetheduck is right. He wrote the latest string-handling logic, of which I wasn't fully aware. You can see it here:

https://github.com/status-im/nim-chronicles/blob/master/chronicles/log_output.nim#L559-L576

Reply:
indeed - it was done this way to avoid inventing yet another text line log format - there's a bunch of implementations of this one already: https://rollout.io/blog/logfmt-a-log-format-thats-easy-to-read-and-write/ so we gain the benefit of the tooling in lots of frameworks and languages built around it

Reply:
It should now exactly match the current behavior. Also, I've moved the "quotation" algorithm to `textlinesserialization.nim` directly and back out of `faststreams` as it is too specific IMO.

I also added variants for type that need not have a  quotation check (such as an `int`) and also added a version that supports `ref Exception` display.